pkg/controller/storagecluster/storagecluster.go

```diff
@@ -752,9 +752,15 @@ func preflightShouldRun(cluster *corev1.StorageCluster) bool {
 				return true
 			}
 
-			// Don't run Vsphere w/PKS
-			if pxutil.IsVsphere(cluster) && preflight.IsPKS() {
-				return false
+			if pxutil.IsVsphere(cluster) {
```

Comment:
Can we please cleanup the logic here? line 751 will be prone to errors. I would recommend that we keep the logic here as the following
```
{
currentVersion := pxutil.GetPortworxVersion(cluster)

if pxutil.IsEKS() && currentVersion.GreaterThanOrEqual(pxVer30) {
return true
}

if pxutil.IsVsphere() && currentVersion.GreaterThanOrEqual(pxVer31){
 if PKS() || localMode{
return false
  }
return true
}

// More platforms here with their respective versions.

// None of the condition matched
return false
}
```